src/test/java/bio/terra/service/dataset/DatasetAzureIntegrationTest.java

```diff
@@ -487,6 +492,118 @@ public void testDatasetFileIngestLoadHistory() throws Exception {
     clearEnvironment();
   }
 
+  @Test
+  public void testDatasetFileRefValidation() throws Exception {
```

Comment:
Looks like a thorough test, nice work! 
Do you know how long it takes to run? I'm guessing we could combine this into one of the existing tests, although those tests are getting rather large 😞

Reply:
It doesn't take all that long to run. I like keeping it in its own test because its testing a specific path.